src/Microsoft.FeatureManagement/FeatureManager.cs

```diff
@@ -209,12 +291,229 @@ await contextualFilter.EvaluateAsync(context, appContext).ConfigureAwait(false)
+
+            if (featureDefinition.Allocation.User != null)
+            {
+                foreach (UserAllocation user in featureDefinition.Allocation.User)
+                {
+                    if (TargetingEvaluator.IsTargeted(targetingContext.UserId, user.Users, _assignerOptions.IgnoreCase))
+                    {
+                        if (string.IsNullOrEmpty(user.Variant))
+                        {
+                            _logger.LogWarning($"Missing variant name for {nameof(featureDefinition.Allocation.User)} allocation in feature {featureDefinition.Name}");
+
+                            return new ValueTask<VariantDefinition>((VariantDefinition)null);
+                        }
+
+                        Debug.Assert(featureDefinition.Variants != null);
+
+                        return new ValueTask<VariantDefinition>(
+                            featureDefinition
+                                .Variants
+                                .FirstOrDefault((variant) => variant.Name == user.Variant));
+                    }
+                }
+            }
+
+            if (featureDefinition.Allocation.Group != null)
+            {
+                foreach (GroupAllocation group in featureDefinition.Allocation.Group)
+                {
+                    if (TargetingEvaluator.IsTargeted(targetingContext.Groups, group.Groups, _assignerOptions.IgnoreCase))
+                    {
+                        if (string.IsNullOrEmpty(group.Variant))
+                        {
+                            _logger.LogWarning($"Missing variant name for {nameof(featureDefinition.Allocation.Group)} allocation in feature {featureDefinition.Name}");
+
+                            return new ValueTask<VariantDefinition>((VariantDefinition)null);
+                        }
+
+                        Debug.Assert(featureDefinition.Variants != null);
+
+                        return new ValueTask<VariantDefinition>(
+                            featureDefinition
+                                .Variants
+                                .FirstOrDefault((variant) => variant.Name == group.Variant));
+                    }
+                }
+            }
+
+            if (featureDefinition.Allocation.Percentile != null)
+            {
+                foreach (PercentileAllocation percentile in featureDefinition.Allocation.Percentile)
+                {
+                    if (TargetingEvaluator.IsTargeted(
+                        targetingContext,
+                        percentile.From,
+                        percentile.To,
+                        _assignerOptions.IgnoreCase,
+                        featureDefinition.Allocation.Seed))
+                    {
+                        if (string.IsNullOrEmpty(percentile.Variant))
+                        {
+                            _logger.LogWarning($"Missing variant name for {nameof(featureDefinition.Allocation.Percentile)} allocation in feature {featureDefinition.Name}");
```

Comment:
The way this is used it seems like it should just be hardcoded literal "percentage"